### DISCUSSION TITLE
How to author content locally?

### DISCUSSION THREAD

**Discussion by ilyabezdelev** [avitorio/outstatic#246] [Q&A] [answered]
We've installed Outstatic on our Next.js website. It's awesome, but I'd like to avoid pushing unfinished posts to `main`.

I want to have a workflow I had with Hugo/Jekyll:
1. Create a branch for a new post
2. Author my content
3. Create a PR
4. Merge approved PR to `main`

This process keeps `main` free of drafts and allows quick local previews of content on each save.

Instead, what I'm doing now - I create a post with Outstatic, pull from main, edit the markdown locally, create a PR, merge to `main` after approval.

Is there a way to use Outstatic with local files?

**Comment by avitorio** [comment, marked as answer]
Hey,
Unfortunately Outstatic doesn't currently support saving to local files.
I'm adding the ability of picking the branch you want to work with in the next version.

**Comment by h34lthbybr33** [comment]
Just want to 2nd this. In addition to keeping the git history to a minimum, I'd also like to call out the metadata doesn't 100% catch up with content changes (especially around deletes). Local would, I think, make this a tad more efficient (given it's essentially a search index file roll-up of the files themselves). Wouldn't need to traverse the branch from afar and commit changes, but instead could monitor/track local. In addition, I could see the "publish" behavior (in "local mode") as the final commit of the content itself and the metadata changes.

Just a thought.

**Reply by avitorio** [reply]
Hey, thanks for the suggestion. Local editing is in the backlog. 
I guess there's more to it than what it appears on the surface.
Making commits from the web interface locally sounds like a good feature though.

**Comment by ilyabezdelev** [comment]
Personally, I would love it if Outstatic could create files and metadata.json entry for a new post without a commit. 

I prefer editing Markdown directly and would love to be able to run a script to create a new post that I could work on locally and publish as a single commit when it's ready.

**Reply by ilyabezdelev** [reply]
As an aside, I find it very inefficient that I have to edit both the post's front matter and metadata.json every time I make a change to the title or description in Markdown.

**Reply by avitorio** [reply]
Hey,
That shouldn't be the case. Do you get any errors in your network tab when saving?

**Reply by ilyabezdelev** [reply]
I'm editing markdown files in VSCode.

I don't use the Outstatic UI except for creating posts. Currently, we only use Outstatic as a database of posts. Once there's support for working on branches, we'll look at using the UI again.
